### PR TITLE
[15.0][ADD] dms_auto_classification: New module

### PR DESCRIPTION
**Pull request by victoralmau**
New module

ejemplo

Please @pedrobaeza and @chienandalu can you review it?

@Tecnativa TT46805

### CODE REVIEW

**Review by chienandalu — APPROVED** [review]
Tested :+1: 

question: is it expected to discard those files not fulfilling the pattern?

**Comment by victoralmau** [comment]
> Tested 👍
> 
> question: is it expected to discard those files not fulfilling the pattern?

Yes, files that do not fulfilling the pattern are not added in details and therefore will not be created.

**Comment by victoralmau** [comment]
Ping @pedrobaeza

File: dms_auto_classification/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2024 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Auto classification files to dms",
```
Comment (pedrobaeza) [inline dms_auto_classification/__manifest__.py]:
```suggestion
    "name": "Auto classify documents into DMS",
```

File: dms_auto_classification/models/dms_classification_template.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Copyright 2024 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class DmsClassificationTemplate(models.Model):
+    _name = "dms.classification.template"
+    _order = "sequence, name"
```
Comment (pedrobaeza) [inline dms_auto_classification/models/dms_classification_template.py]: Always put `id` at the end

```suggestion
    _order = "sequence,name,id"
```

File: dms_auto_classification/security/security.xml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo>
+    <record id="rule_multi_company_dms_classification_template" model="ir.rule">
+        <field name="name">DMS Classification Template multi-company</field>
+        <field name="model_id" ref="model_dms_classification_template" />
+        <field name="global" eval="True" />
+        <field
+            name="domain_force"
+        >['|',('company_id','=',False),('company_id','in',company_ids)]</field>
```
Comment (pedrobaeza) [inline dms_auto_classification/security/security.xml]:
```suggestion
        >[('company_id', 'in', [False] + company_ids)]</field>
```

File: dms_auto_classification/static/description/icon.png
Comment (pedrobaeza) [inline dms_auto_classification/static/description/icon.png:1]: Take this icon instead (SVG and PNG included):

icon
icon

File: dms_auto_classification/wizards/wizard_dms_classification_views.xml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+                        attrs="{'invisible' : [('state', '!=', 'draft')]}"
+                    />
+                    <button
+                        name="action_classify"
+                        string="Classify"
+                        type="object"
+                        class="btn-primary"
+                        attrs="{'invisible' : [('state', '!=', 'analyze')]}"
+                    />
+                    <button string="Cancel" class="btn-secondary" special="cancel" />
```
Comment (pedrobaeza) [inline dms_auto_classification/wizards/wizard_dms_classification_views.xml:53]: On the last stage, hide this button and put another with the text "Close" instead, or better, directly navigate to a list view with the created files (and thus, remove the last stage).

**Comment by victoralmau** [comment]
Changes done.

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**